cmd/plan.go

```diff
@@ -162,7 +169,72 @@ func planInstall(rv cran.RVersion) (*cran.PkgDb, gpsr.InstallPlan) {
 	for pkg := range ip.DepDb {
 		pkgs = append(pkgs, pkg)
 	}
+
 	log.Infoln("total packages required:", len(ip.StartingPackages)+len(ip.DepDb))
 	log.Infoln("resolution time", time.Since(startTime))
 	return cdb, ip
 }
+
+
+type InstalledPackage struct {
+	Name string
+	Version string
+	Repo string
+}
+
+func GetPriorInstalledPackages(fileSystem afero.Fs, libraryPath string) map[string]InstalledPackage {
+
+	installed := make(map[string]InstalledPackage)
+
+	installedLibrary, err := fileSystem.Open(libraryPath)
+
+	if err != nil {
+		//panic?
+		return installed
+	}
+
+	installedPkgs, _ := installedLibrary.Readdir(0)
+
+	for _, f := range installedPkgs {
+		descriptionFilePath := filepath.Join(libraryPath, f.Name(), "DESCRIPTION")
+		descriptionFile, err := fileSystem.Open(descriptionFilePath)//, _ := fs.Open()
+
+		if err != nil {
+			//panic?
+			log.WithField("file", descriptionFilePath).Warn("DESCRIPTION missing from installed package")
+		}
+		scanner := bufio.NewScanner(descriptionFile)//#bufio.Scanner(fs.Open(f))
+		log.WithField("description file", descriptionFilePath).Debug("scanning DESCRIPTION file")
+
+		var pkgName, pkgVersion, pkgRepo = "", "", ""
+		for scanner.Scan() {
+			splitLine := strings.Split(scanner.Text(), ":")
+			switch strings.ToLower(splitLine[0]) {
+			case "package":
+				pkgName = strings.TrimSpace(splitLine[1])
+			case "version":
+				pkgVersion = strings.TrimSpace(splitLine[1])
+			case "repository":
+				pkgRepo = strings.TrimSpace(splitLine[1])
+			default:
+				log.WithField("line", scanner.Text()).Debug("no info found on line")
```

Comment:
there will be a lot of additional info found so this debug log will be huge (namely all non pkg/version/repo lines in every description file)